server/src/modules/notifications/agents/gotify.ts

```diff
@@ -0,0 +1,104 @@
+import axios from 'axios';
+import { hasNotificationType } from '../notifications.service';
+import type { NotificationAgent, NotificationPayload } from './agent';
+import { Logger } from '@nestjs/common';
+import { SettingsService } from '../../settings/settings.service';
+import {
+  NotificationAgentConfig,
+  NotificationAgentKey,
+  NotificationType,
+} from '../notifications-interfaces';
+import { Notification } from '../entities/notification.entities';
+
+interface GotifyPayload {
+  title: string;
+  message: string;
+  priority: number;
+  extras: Record<string, unknown>;
+}
+
+class GotifyAgent implements NotificationAgent {
+  public constructor(
+    private readonly appSettings: SettingsService,
+    private readonly settings: NotificationAgentConfig,
+    readonly notification: Notification,
+  ) {
+    this.notification = notification;
+  }
+
+  private readonly logger = new Logger(GotifyAgent.name);
+
+  getNotification = () => this.notification;
+
+  getSettings = () => this.settings;
+
+  getIdentifier = () => NotificationAgentKey.GOTIFY;
+
+  public shouldSend(): boolean {
+    const settings = this.getSettings();
+
+    if (settings.enabled && settings.options.url && settings.options.token) {
+      return true;
+    }
+
+    return false;
+  }
+
+  private getNotificationPayload(
+    type: NotificationType,
+    payload: NotificationPayload,
+  ): GotifyPayload {
+    const priority = 0;
+
+    const title = payload.event
+      ? `${payload.event} - ${payload.subject}`
+      : payload.subject;
+    let message = payload.message ?? '';
+
+    for (const extra of payload.extra ?? []) {
+      message += `\n\n**${extra.name}**\n${extra.value}`;
+    }
+
+    return {
+      extras: {
+        'client::display': {
+          contentType: 'text/markdown',
+        },
+      },
+      title,
+      message,
+      priority,
+    };
+  }
+
+  public async send(
+    type: NotificationType,
+    payload: NotificationPayload,
+  ): Promise<boolean> {
+    const settings = this.getSettings();
+
+    if (
+      !payload.notifySystem ||
```

Comment:
What's notifySystem used for? From what I can see it's always set to false so this and a few other agents will always return early.

Reply:
That's a leftover i still need to remove. I copied the notification agents from Overseerr and adapted them to something Nestjs understands. I believe they use it for notifications regarding the Overseerr application itself (like errors).